### PR TITLE
*: enable column-type-change totally (#25263)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #25263 to release-5.1
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/25280
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/25280:release-5.1-ac06d78691be
```

---

### What problem does this PR solve?

Remove tidb_enable_change_column_type totally.

### What is changed and how it works?

Remove tidb_enable_change_column_type totally.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- No release note

### CODE REVIEW

File: ddl/column_test.go [hunk 1]
```diff
@@ -1159,6 +1159,10 @@ func (s *testColumnSuite) TestModifyColumn(c *C) {
 		WithLease(testLease),
 	)
 	ctx := testNewContext(d)
+<<<<<<< HEAD
+=======
+
```
Comment (AilinKid) [inline ddl/column_test.go:1164]: @wjhuang2016

**Review by tangenta — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Pull request merged**